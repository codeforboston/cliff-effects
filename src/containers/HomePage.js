import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class HomePage extends Component {
  render() {
    return (
      <div id='HomePage'>
        <div id='HomeContent'>
          <div className="main-wrapper">
            <div className="headings">
              <h1>Cliff Effects Tool</h1>
              <h2>GUIDANCE PROTOTYPE*</h2>
            </div>

            <div className='center-contents'>
              <Link
                className='home-button hb-left'
                to='/visit/54321/1'>
                Get Started
              </Link>

              <Link
                className='home-button hb-right'
                to='/about'>
                Learn More
              </Link>
            </div>
            <p className="home-disclaimer">*This is a prototype and should not be used to make financial decisions.</p>
          </div>
        </div>
      </div>
    );
  }
}

export default HomePage;
