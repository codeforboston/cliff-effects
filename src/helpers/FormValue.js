/** Handles the core of form values, including:
 * - Storing both a current and future value
 * - Tracking when the user manipulates a future value
 * - Propagating current values to future values if they
 * haven't been manipulated
 * - Making this instance serializable
 * 
 * All values stored in a FormValue instance should be valid
 * 
 * There must be a better way to write that.
 * 
 * @class
 * @param {string} name - the name of the client property
 * @param {any} val - the initial value
 * @param {function} [makeValid] - transforms a given value into
 * a valid value
 * 
 * @returns FormValue
 * @todo Finish description of FormValue return value
 */
class FormValue {

  constructor ( name, val, makeValid ) {

    var formVal = this;

    formVal.name = name;
    // Is there a function to set this, or just `formVal.futureChanged = true;`?
    formVal.futureChanged = false;
    // Not sure about providing this default...
    formVal.makeValid = makeValid || function ( val ) { return val; };

    formVal.setCurrent( val );

  }  // End FormValue.constructor()


  setCurrent ( val ) {

    var formVal = this;

    val = formVal.makeValid( val, 'current', formVal );

    formVal.current = val;
    if ( !formVal.futureChanged ) {
      formVal.setFuture( val )
    }

    return formVal;

  }  // End FormValue.setCurrent()


  setFuture ( val ) {
    
    var formVal = this;

    val = formVal.makeValid( val, 'future', formVal );
    formVal.future = val;
    
    return formVal;

  }  // End FormValue.setFuture()


  serialize () {

    var formVal = this;

    var obj     = {};
    obj.current = formVal.current;
    obj.future  = formVal.future;
    obj.futureChanged = formVal.futureChanged;

    return obj;
  }  // End FormValue.serialize()


  deserialize ( obj, shouldSetFuture ) {

    var formVal = new FormValue( obj.name, obj.makeValid );

    formVal.futureChanged = Boolean( obj.futureChanged );
    formVal.setCurrent( obj.current || null );

    if ( shouldSetFuture ) {
      formVal.setFuture( obj.future || null );
    }

    return formVal;

  }  // End FormValue.deserialize()

};  // End FormValue{}


export { FormValue };
