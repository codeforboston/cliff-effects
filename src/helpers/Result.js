/**
* All benefit programs' return value should be an instance of Result class.
*
* @todo Implement `.expirationDate`to track  when data needs to be update
* @external
* @class
* @param {object}
* @param {string} trial.result - 'good', 'warning', or 'bad' are the proposed
* three values, but right now they're 'good', 'information', and 'warning'
* @param {string} trial.details - Explanation of results that will be seen
* by the user. Examples: 'All good!', 'Because your income is x% above some
* value, your benefit amount could go down soon.'
* @param {string} trial.benefitValue - Monthly subsidy benefit will provide.
* @param {object} trial.data -  other information if needed. The code using your `.data` will have to know
* what to expect in there.
*/

class Result {
  constructor ( trial ) {
    this.result       = trial.result;
    this.details      = trial.details;
    this.benefitValue = trial.benefitValue;
    this.data         = trial.data;
  }
};

export { Result }
