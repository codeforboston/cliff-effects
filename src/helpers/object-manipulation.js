/** For object manipulation. Transformation? */

/** 
* Add the properties from needy to giver, overwriting existing
* properties of needy if there's a conflict.
* 
* @todo Change name so it doesn't sound like destroys both objects.
*/
const merge = function ( needy, giver ) { // Function to merge all of the properties from one object into another
    for ( var key in giver ) {  needy[key] = giver[key]; }
    return  needy;
};

/** Make the first letter of a word capitalized. */
const capitalizeWord = function ( word ) {
	return word[ 0 ].toUpperCase() + word.substr( 1 );
}


export { merge, capitalizeWord };
