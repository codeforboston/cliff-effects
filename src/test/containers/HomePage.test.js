import React from 'react';
import { mount } from 'enzyme';

import HomePage from '../../containers/HomePage';
import { withRouter } from '../helpers';

describe('<HomePage>', () => {
  it('matches snapshot', () => {
    expect(mount(withRouter(<HomePage />))).toMatchSnapshot();
  });
});
