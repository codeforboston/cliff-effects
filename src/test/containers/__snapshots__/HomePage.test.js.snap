// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage> matches snapshot 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
        "pathname": "/",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <HomePage>
      <div
        id="HomePage"
      >
        <div
          id="HomeContent"
        >
          <div
            className="main-wrapper"
          >
            <div
              className="headings"
            >
              <h1>
                Cliff Effects Tool
              </h1>
              <h2>
                GUIDANCE PROTOTYPE*
              </h2>
            </div>
            <div
              className="center-contents"
            >
              <Link
                className="home-button hb-left"
                replace={false}
                to="/visit/54321/1"
              >
                <a
                  className="home-button hb-left"
                  href="/visit/54321/1"
                  onClick={[Function]}
                >
                  Get Started
                </a>
              </Link>
              <Link
                className="home-button hb-right"
                replace={false}
                to="/about"
              >
                <a
                  className="home-button hb-right"
                  href="/about"
                  onClick={[Function]}
                >
                  Learn More
                </a>
              </Link>
            </div>
            <p
              className="home-disclaimer"
            >
              *This is a prototype and should not be used to make financial decisions.
            </p>
          </div>
        </div>
      </div>
    </HomePage>
  </Router>
</MemoryRouter>
`;
